# Exam 1 Answers Sara Temino Macaya

# Problem 2


In [2]:
ls -lh

total 2.4M
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  6 05:40  conv.txt
-rw-rw-r-- 1 osboxes osboxes  147 Sep  6 05:40  DISCLAIMER
-rw-rw-r-- 1 osboxes osboxes  509 Sep  6 10:09  Exam_1_Answers.ipynb
-rw-rw-r-- 1 osboxes osboxes 5.5K Sep  6 05:40 'Exam Week 1.ipynb'
-rw-rw-r-- 1 osboxes osboxes 3.9K Sep  6 05:40 'Exam Week 2.ipynb'
-rw-rw-r-- 1 osboxes osboxes 4.2K Sep  6 05:40  Germplasm.tsv
-rw-rw-r-- 1 osboxes osboxes  141 Sep  6 05:40  how_to_convert.txt
-rw-rw-r-- 1 osboxes osboxes 1.1K Sep  6 05:40  LICENSE
-rw-rw-r-- 1 osboxes osboxes  637 Sep  6 05:40  LocusGene.tsv
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  6 05:40  Locus_Germplasm_Phenotype_20130122.txt
-rw-rw-r-- 1 osboxes osboxes  743 Sep  6 05:40  README.md


## ls -lh prints the directory content at human readable size
### Users : "rw" read and write
### Groups: "rw" read and write
### Anyone: "r" only read

# Problem 3

In [5]:
head -1 Locus_Germplasm_Phenotype_20130122.txt

Locus_name)	Germplasm_name	phenotype	pubmed_id


## Head -1 prints only the first line

# Problem 4

In [7]:
wc -l Locus_Germplasm_Phenotype_20130122.txt

7216 Locus_Germplasm_Phenotype_20130122.txt


## wc -l counts the total number of lines. In this case, the file has 7216 lines

# Problem 5 

In [9]:
tail +2 Locus_Germplasm_Phenotype_20130122.txt > Data_Only.csv

In [10]:
wc -l Data_Only.csv

7215 Data_Only.csv


## since all lines except the first one have been selected, there is one line less, corroborating the effectiveness of the command

# Problem 6

In [ ]:
grep -ia "root" Locus_Germplasm_Phenotype_20130122.txt

## Option -i makes the search insensitive to cases. Like I don not know the original source of this file, i use -a.

# Problem 7

In [1]:
grep -ia "root" Locus_Germplasm_Phenotype_20130122.txt | grep -E -ioa 'AT[1-5]G[0-9]{5}' > Root-associated-Loci.txt

## Using the same command line that the one I used in the previous exersise and then a pipe to a grep command that print the AGI Locus Codes (-o option to print the matches) from every line that contains the word root using them regular expression "at[1-5]g[0-9]{5}" because the number of the chromosome can be only between 0 and 9. Finally, I redirect the standard output to the file Root-associated-loci.txt

# Problem 8

In [2]:
grep -ia "root" Locus_Germplasm_Phenotype_20130122.txt | grep -E -oa "[0-9]+$" > Root-associated-Publications.txt


## First I filter the phenotypes as I did in the previous question (those which include the word "root"), and then I use grep in order to find the PubMed ID. The length of the ID is unknown so I use the "+" character. Finally, I use the regular expression "$" to find the IDbecause they appear at the end of the line.

# Problem 9

In [22]:
grep -ia "root" Locus_Germplasm_Phenotype_20130122.txt | grep -E -o '[1-5]'| sort | uniq -c

   3373 1
   1978 2
   1746 3
   1242 4
   1621 5


## As the genes of interest are those associated with root, I use the grep command for select these. To avoid the repeated chromosomes I use the command uniq with the argument -c to print the numbers of genes that are in each chromosome. To use the command uniq is necessary apply too the command sort.

This work has been done with Miguel Rey Romero because we are a couple and we live together. We have also delivered it late because we were travelling from Benidorm to Madrid, and there has been a cut in the A3 road which has delayed us a little more. We are sorry for the inconvenience. Thank you for your aŧtention.